src/main/java/com/spirnt/mission/discodeit/service/basic/BasicChannelService.java

```diff
@@ -0,0 +1,148 @@
+        if(readStatus!=null) readStatusService.update(readStatus.getId(), new ReadStatusDto(userId, channelId,Instant.now()));
+        // UserStatus 업데이트 -> 온라인 && 현재 활동 중으로 간주
+        userStatusService.updateByUserId(userId, new UserStatusUpdate(UserStatusType.ONLINE, Instant.now()));
+
+        return new ChannelResponse(channel, participantIds, lastMessageAt);
+    }
+
+    @Override
+    public List<ChannelResponse> findAllByUserId(UUID userId) {
+        // UserStatus 업데이트 -> 온라인 && 현재 활동 중으로 간주
+        userStatusService.updateByUserId(userId, new UserStatusUpdate(UserStatusType.ONLINE, Instant.now()));
+        Map<UUID, Channel> data = channelRepository.findAll();
+        return data.values().stream()
+                //PUBLIC이거나 User가 참여한 PRIVATE 채널이거나
+                //참여 여부는 ReadStatus 존재 여부로 확인
+                .filter(channel -> (channel.getType() == ChannelType.PUBLIC) ||
+                        (channel.getType() == ChannelType.PRIVATE && readStatusService.existsByUserIdChannelId(userId, channel.getId())))
+                .sorted(Comparator.comparing(channel -> channel.getCreatedAt()))
+                // ChannelResponse로 mapping
+                .map(channel -> find(userId, channel.getId()))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public Channel update(UUID channelId, ChannelUpdateRequest channelUpdateRequest) {
+        Channel channel = channelRepository.findById(channelId)
+                .orElseThrow(() -> new NoSuchElementException("Channel ID: " + channelId + " Not Found"));
+        // PRIVATE 채널은 업데이트할 수 없음
+        if (channel.getType() == ChannelType.PRIVATE) {
+            throw new IllegalStateException("Private Channel Cannot be updated");
+        }
+        channel.update(channelUpdateRequest.getName(), channelUpdateRequest.getDescription());
+        channelRepository.save(channel);
+        return channel;
+    }
+
+    @Override
+    public void delete(UUID channelId) {
+        if(!channelRepository.existsById(channelId)) {
+            throw new NoSuchElementException("Channel ID: " + channelId + " Not Found");
+        }
+        // 채널에 속한 메세지 삭제
+        Map<UUID, Message> data = messageRepository.findAll();
+        if (data == null) return;
+        data.values().stream()
+                .forEach(message -> {
+                    // 해당 채널에 쓰여진 메세지 객체를 찾고
+                    if (message.getChannelId().equals(channelId)) {
+                        // 그 객체를 레포지토리에서 삭제
+                        messageRepository.delete(message.getId());
+                    }
+                });
+        // 채널과 관련된 ReadStatus 삭제
+        readStatusService.deleteByChannelId(channelId);
+        // 채널 삭제
+        channelRepository.delete(channelId);
+    }
+
+
+    // 해당 채널 메세지를 정렬하여 가장 최근 메세지 시간 찾기
+    public Optional<Instant> findLastMessageInChannel(UUID channelId) {
+        List<Message> messages = messageRepository.findAll().values().stream().toList();
```

Comment:
```suggestion
        List<Message> messages = messageRepository.findAll().values();
```
이렇게 해도 동일하게 동작할거 같네요!

아래 메소드도 추가하면 어때요? 
보니까 userStatusRepository.findAllByChannelId 처럼 잘 활용될 수 있을거 같아요!
```suggestion
        List<Message> messages = messageRepository.findAllByChannelId();
```